### PR TITLE
Remove issued_at value check

### PR DESCRIPTION
**Pull request by lbalmaceda**
### Changes

The ID Token `iat` claim value check should be optional.

### Testing

Please describe how this can be tested by reviewers. Be specific about anything not tested and reasons why. Since this library has unit testing, tests should be added for new functionality and existing tests should complete without errors.

- [x] This change adds unit test coverage

- [x] This change adds integration test coverage

- [x] This change has been tested on the latest version of the platform/language or why not

### Checklist

- [x] I have read the [Auth0 general contribution guidelines](https://github.com/auth0/open-source-template/blob/master/GENERAL-CONTRIBUTING.md)

- [x] I have read the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md)

- [x] All existing and new tests complete without errors

### CODE REVIEW

File: auth0/src/main/java/com/auth0/android/provider/IdTokenVerifier.java [hunk 1]
```diff
@@ -68,14 +68,6 @@ void verify(@NonNull JWT token, @NonNull IdTokenVerificationOptions verifyOption
             throw new TokenValidationException("Issued At (iat) claim must be a number present in the ID token");
         }
 
-        cal.setTime(token.getIssuedAt());
```
Comment (damieng) [inline auth0/src/main/java/com/auth0/android/provider/IdTokenVerifier.java]: Having a single mutable calendar object and then setting it to different values betweet tests seems quite an unusual pattern. Is this normal in Java?

Comment (lbalmaceda) [reply auth0/src/main/java/com/auth0/android/provider/IdTokenVerifier.java]: Think of Calendar as a "dates calculator". It's easier to do _math_ operations on a Calendar than a Date instance. Note though that once the math is done, an instance of Date is obtained from that Calendar for comparison against another one.

Comment (damieng) [reply auth0/src/main/java/com/auth0/android/provider/IdTokenVerifier.java]: I get why the Calendar object is being used - it's the re-use for multiple calculations that seems odd.

Comment (lbalmaceda) [reply auth0/src/main/java/com/auth0/android/provider/IdTokenVerifier.java]: I've always seen it being used like this, at least for when the min JDK is 7. `Calendar` is already a heavy object. There's no harm in re-using it and is better than creating multiple instances.

**Pull request merged**